### DISCUSSION TITLE
Error Threshold Management

### DISCUSSION THREAD

**Discussion by chloeminkyung** [getdozer/dozer#620] [Feature Requests]
This discussion about error threshold management is about:
- putting some buffer to Dozer pipeline not to be panic killed with every single error it encounters
- error here refers to internal error

### How to handle?
- [x] Adding a flag in Dozer CLI `app run` for stating no. of errors 
- [x] Setting up default no. of errors to be buffered (e.g. 10)
- [ ] [Leveling different kinds of error (in case of invasive self-distructive error, should we still panic kill the pipeline?)](https://github.com/getdozer/dozer/discussions/620#discussioncomment-5599362)

**Comment by chloeminkyung** [comment]
error happens during build stage -> critical (OrchestrationError unless it's forwarding ExecutionError)
error happens during process stage -> non-critical (ExecutionError)

- [ ] categorize errors into `critical` and `non-critical`
- [ ] refactor unused error variants
- [ ] factory is returning `ExecutionError` -> better to specify as in `FactoryError`

**Comment by chubei** [comment]
Should error threshold be in the config file?
